tools/build-tools/src/typeValidator/packageJson.ts

```diff
@@ -58,33 +86,59 @@ export function getPackageDetails(packageDir: string): PackageDetails {
             pkgJson.version;
 
     // if the packages has no type validation data, then initialize it
-    if(pkgJson.typeValidation === undefined){
-        pkgJson.typeValidation = {
-            version: normalizedVersion,
-            broken: {}
+   if(updateOptions !== undefined && normalizedVersion !== pkgJson.typeValidation?.version){
+        /*
+         * this is where we build the previous version we use to compare against the current version.
+         * For both major and minor we use semver such that the current major is compared against the latest
+         * previous major, and the same for minor.
+         * For patch we do not use semver, we compare directly to the previous patch version.
+         *
+         * We do this to align with our release process. We are strictest between patches, and looser between minor and major
+         * We may need to adjust this as we adjust our release processes.
+         */
+        let normalizeParts = normalizedVersion.split(".").map((p)=>Number.parseInt(p));
+        const validationVersionParts = pkgJson.typeValidation?.version?.split(".").map((p)=>Number.parseInt(p)) ?? [0,0,0];
+        let semVer="^"
+        if(normalizeParts[0] !== validationVersionParts[0]){
+            normalizeParts= [normalizeParts[0] -1, 0, 0];
+        }else if(normalizeParts[1] !== validationVersionParts[1]){
+            normalizeParts= [normalizeParts[0], normalizeParts[1] -1, 0];
```

Comment:
how does this work for the virtualPatch minor versions right now?

Reply:
it doesn't account for it specifically, but it doesn't really matter as we treat major and minor the same, which are really the only cases where semver is better

Reply:
i guess you could say that by have special minor treatment i am handling it just like we handle major.